components/ProductCard.js

```diff
@@ -1,8 +1,29 @@
 import styled, { css } from "styled-components";
 import MoreButton from "./MoreButton";
 import Link from "next/link";
+import DeleteButton from "./DeleteButton";
+import { useRouter } from "next/router";
+import { mutate } from "swr";
 
 export default function ProductCard({ name, quantity, category, _id }) {
+  const router = useRouter();
+  async function handleDelete() {
+    const confirmed = window.confirm(
+      `Are you sure you want to delete "${name}"?`
+    );
+    if (!confirmed) return;
+
+    const response = await fetch(`/api/shoppinglist/${_id}`, {
+      method: "DELETE",
+    });
+    if (response.ok) {
+      mutate("/api/shoppinglist", (currentData) =>
+        currentData.filter((item) => item._id !== _id)
+      );
```

Comment:
you don't have to pass the second argument to mutate, unless you want to use so called "optimistic" UI updates (updating the display of your App before actually changing the data in the database). Instead, you can just call mutate and pass the url string like you did here to instruct swr to refetch the respective data. This is call "revalidating".